src/main/java/seedu/address/model/person/Category.java

```diff
@@ -0,0 +1,81 @@
+package seedu.address.model.person;
+
+import static java.util.Objects.requireNonNull;
+import static seedu.address.commons.util.AppUtil.checkArgument;
+
+import java.util.HashMap;
+
+/**
+ * Represents an Eatery's category in the EatMe application.
+ */
+public class Category {
+
+    public static final String MESSAGE_CONSTRAINTS =
+            "Categories should only contain alphabets and spaces, and it should not be blank";
+
+    /*
+     * The first character of the address must not be a whitespace,
+     * otherwise " " (a blank string) becomes a valid input.
+     */
+    public static final String VALIDATION_REGEX = "([A-Za-z]\\w+[ A-Za-z]*)";
+
+    private static int categoryId = -1;
+    private static HashMap<String, Category> categories = new HashMap<String, Category>();
+
+    private int id;
+    private String name;
+
+    /**
+     * Constructs a {@code Category}.
+     *
+     * @param name A valid category name.
+     */
+    public Category(String name) {
+        requireNonNull(name);
+        checkArgument(isValidCategory(name), MESSAGE_CONSTRAINTS);
+        categoryId = categoryId + 1;
+        this.id = categoryId;
+        this.name = name;
+    }
+
+    /**
+     * Returns a new Category object if Category with the same name doesn't exist,
+     * else it returns the existing Category object.
+     */
+    public static Category create(String name) {
+        Category cat = categories.get(name);
```

Comment:
Are the categories being added to the hashmap when they are not present?

Reply:
yeah like if this is an entirely new category e.g. categories hashmap currently only has the "chinese" category and your new category is "western", then it'll check the categories hashmap if there's an entry and if not, it'll add it in? you don't add in a category that already exists into the hashmap bc like we don't want duplicates, right